content/intro-to-storybook/ember/en/composite-component.md

```diff
@@ -0,0 +1,274 @@
+
+![Qunit logo](/intro-to-storybook/logo-qunit.jpg)
+
+### Unit tests with Qunit
+
+Storybook stories paired with manual visual tests go a long way to avoiding UI bugs. If stories cover a wide variety of component use cases, and we use tools that ensure a human checks any change to the story, errors are much less likely.
+
+However, sometimes the devil is in the details. A test framework that is explicit about those details is needed. Which brings us to unit tests.
+
+In our case, we want our `TaskList` to render any pinned tasks **before** unpinned tasks that it has passed in the `tasks` prop. Although we have a story (`WithPinnedTasks`) to test this exact scenario, it can be ambiguous to a human reviewer that if the component **stops** ordering the tasks like this, it is a bug. It certainly won’t scream **“Wrong!”** to the casual eye.
+
+So, to avoid this problem, we can use Qunit to render the component and run some DOM querying code to verify salient features of the output.
+
+Create a test file called `tests/integration/task-list-test.js`. Here, we’ll build out our tests that make assertions about the output.
+
+```javascript
+// tests/integration/task-list-test.js
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+module('Integration | Component | TaskList', function(hooks) {
+  setupRenderingTest(hooks);
+  const taskData = {
+    id: '1',
+    title: 'Test Task',
+    state: 'TASK_INBOX',
+    updatedAt: new Date(2018, 0, 1, 9, 0),
+  };
+  const tasklist = [
+    { ...taskData, id: '1', title: 'Task 1' },
+    { ...taskData, id: '2', title: 'Task 2' },
+    { ...taskData, id: '3', title: 'Task 3' },
+    { ...taskData, id: '4', title: 'Task 4' },
+    { ...taskData, id: '5', title: 'Task 5' },
+    { ...taskData, id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
+  ];
+
+  test('renders pinned tasks at the start of the list', async function(assert) {
+    this.set('tasks', tasklist);
+    this.set('testpinTask', actual => {
+      let expected = 1;
+      assert.deepEquals(actual, expected);
+    });
+
+    this.set('testarchiveTask', actual => {
+      let expected = 1;
+      assert.deepEquals(actual, expected);
+    });
+
+    await render(
+      hbs`{{TaskList tasks=tasks pinTask=(action testpinTask) archiveTask=(action testarchiveTask) }}`
+    );
+    assert.dom(this.element.querySelector('.list-item:nth-of-type(1)')).hasClass('TASK_PINNED');
+  });
+});
+```
+
+![TaskList test runner](/intro-to-storybook/tasklist-testrunner-ember.png)
+
+Contrary to the other versions of this tutorial, with ember we can't import the data and stories used in our story file created earlier, without introducing a lot of complexity and that's beyond the scope of the tutorial. For now we'll copy over the values used in the story file to help out with our tests.
```

Comment:
Capitalize Ember

Reply:
It was fixed with the latest commit. Going to mark it as resolved if you don't mind.